### PR TITLE
[Tests] Add attentions_option to ModelTesterMixin

### PR DESCRIPTION
**Pull request by NielsRogge**
# What does this PR do?

The library is called HuggingFace Transformers, I know.. but we recently have additions of non-Transformer based models, that don't use attention (namely ConvNeXT and PoolFormer). Soon, we'll also have ResNet in the library (which will back Transformer-based models such as DETR).

As these models don't use attention, they need to overwrite 3 tests:
* `test_attention_outputs`
* `test_retain_grad_hidden_states_attentions`
* `test_model_outputs_equivalence`

This PR adds a `has_attentions` attribute to `ModelTesterMixin` which can be set to False for such models. It will then make sure these tests are properly tested, without taking into account attention.

### CODE REVIEW

**Review by FrancescoSaverioZuppichini — COMMENTED** [review]
Thanks! I am not super familiar with this codebase so I am not going to approve it but LGTM!

**Review by LysandreJik — APPROVED** [review]
LGTM, thank you @NielsRogge 

Pinging @ydshieh as well for second review

**Comment by LysandreJik** [comment]
You'll need to fix the failing tests as well.

**Comment by ydshieh** [comment]
Thanks for this work @NielsRogge !

I left a review comment (question) about `test_forward_signature`.

And then a minor question here:

https://github.com/huggingface/transformers/blob/114b62cc9927d85804540d46a6f85e57ba13ea51/tests/test_modeling_common.py#L1046-L1049

It is a bit strange (to me) to have `config.output_attentions = True` without any condition like `if self.has_attentions:`. WDYT?

File: tests/test_modeling_common.py [hunk 1]
```diff
@@ -1040,7 +1045,7 @@ def check_hidden_states_output(inputs_dict, config, model_class):
     def test_retain_grad_hidden_states_attentions(self):
         config, inputs_dict = self.model_tester.prepare_config_and_inputs_for_common()
         config.output_hidden_states = True
-        config.output_attentions = True
+        config.output_attentions = True if self.has_attentions else False
```
Comment (NielsRogge) [inline tests/test_modeling_common.py:1048]: @ydshieh I've taken your comment into account, is this OK?

**Comment by NielsRogge** [comment]
@LysandreJik I've fixed the failing tests, failing test seems unrelated

File: tests/test_modeling_common.py (hunk 1 above)
Comment (ydshieh) [reply tests/test_modeling_common.py:1048]: LGTM, thanks!

**Review by ydshieh — APPROVED** [review]
LGTM!

File: tests/poolformer/test_modeling_poolformer.py [hunk 2]
```diff
@@ -150,100 +151,6 @@ def test_inputs_embeds(self):
...
-
-            tuple_inputs = self._prepare_for_class(inputs_dict, model_class)
-            dict_inputs = self._prepare_for_class(inputs_dict, model_class)
-            check_equivalence(model, tuple_inputs, dict_inputs, {"output_hidden_states": True})
-
-            tuple_inputs = self._prepare_for_class(inputs_dict, model_class, return_labels=True)
-            dict_inputs = self._prepare_for_class(inputs_dict, model_class, return_labels=True)
-            check_equivalence(model, tuple_inputs, dict_inputs, {"output_hidden_states": True})
-
-    def test_forward_signature(self):
```
[the remark this answers is not in the document]
Comment (ydshieh) [reply tests/poolformer/test_modeling_poolformer.py:231]: I saw you put it back, ignore this review (I forgot to submit the review previously)

[the remark this answers is not in the document]
Comment (NielsRogge) [reply tests/poolformer/test_modeling_poolformer.py:231]: Ok!

**Comment by LysandreJik** [comment]
Feel free to merge if every test passes after a rebase

**Pull request merged**